packages/block-editor/src/hooks/block-styles.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { createHigherOrderComponent, compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { Fragment } from '@wordpress/element';
+import { hasBlockSupport, getBlockType } from '@wordpress/blocks';
+import { PanelBody } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../components/inspector-controls';
+import BlockStyles from '../components/block-styles';
+import DefaultStylePicker from '../components/default-style-picker';
+
+export const withInspectorControls = createHigherOrderComponent(
+	compose( [
+		withSelect( ( select ) => {
+			const { getSelectedBlockClientId, getBlockName } = select(
+				'core/block-editor'
+			);
+			const { getBlockStyles } = select( 'core/blocks' );
+			const selectedBlockClientId = getSelectedBlockClientId();
+			const selectedBlockName =
+				selectedBlockClientId && getBlockName( selectedBlockClientId );
+			const blockType =
+				selectedBlockClientId && getBlockType( selectedBlockName );
+			const blockStyles =
+				selectedBlockClientId && getBlockStyles( selectedBlockName );
+			return {
+				blockType,
+				selectedBlockClientId,
+				hasBlockStyles: blockStyles && blockStyles.length > 0,
+			};
+		} ),
```

Comment:
An opportunity to refactor to use useSelect?

Reply:
Yes, for sure. Why do you prefer using hooks instead of composition?

Reply:
It avoids an unnecessary block wrapper and I believe it results in a better API as you know exactly what is an external API of the component (prop) and what is just internal.

Reply:
Awesome. Thanks a lot. Addressed 3cb6621233384b833c2e276a7f1f271eff493355